src/lib.rs

```diff
@@ -28,10 +28,16 @@
 // https://people.csail.mit.edu/rivest/Md5.c
 // https://tools.ietf.org/html/rfc1321
 
-use std::convert;
-use std::fmt;
-use std::io;
-use std::ops;
+#![cfg_attr(not(any(test, feature = "std")), no_std)]
```

Comment:
Why do you want to have `test`? Probably one should change [.travis.yaml](https://github.com/stainless-steel/md5/blob/master/.travis.yml#L10) to run everything with and without the `std` feature. What do you think?

Reply:
I guess test isn't technically needed. Testing the two seperately doesn't make much sense as testing the std feature, already tests all the no_std code (cargo features are additive).

These changes effectively turn the crate into a no_std crate, that by default, opts into std features, does that make more sense?

Reply:
I mean, some `#![…]` might be misplaced, and it won’t work once the feature is disabled. It’s not a check for the written tests, but for being able to compile without the feature.

Reply:
That makes sense, I've added a no_std build to travis.